compiler/hieFile/HieAst.hs

```diff
@@ -816,10 +816,13 @@ instance ( a ~ GhcPass p
         [ toHie $ C Use name
         , toHie $ RC RecFieldAssign $ binds
         ]
-      RecordUpd {rupd_expr = expr, rupd_flds = upds}->
+      RecordUpd {rupd_expr = expr, rupd_flds = Left upds}->
         [ toHie expr
         , toHie $ map (RC RecFieldAssign) upds
         ]
+      RecordUpd {rupd_expr = expr, rupd_flds = Right _ }->
+        [ toHie expr
```

Comment:
I wonder if this is a bug in the upstream implementation. Afaict, this is either broken by dropping the field or unused and should be a panic. I think it’s the second.